tests/ion_schema_tests_runner.rs

```diff
@@ -0,0 +1,167 @@
+    let mut schema_system = SchemaSystem::new(vec![Box::new(authority)]);
+
+    for path in SCHEMA_TEST_LIST {
+        print!("Reading {}... ", path);
+
+        // get the schema content from given schema file path
+        let ion_content = fs::read(path).unwrap();
+        let iterator = element_reader().iterate_over(&ion_content).unwrap();
+        let schema_content = iterator
+            .collect::<Result<Vec<OwnedElement>, IonError>>()
+            .unwrap();
+
+        let type_store = &mut TypeStore::new();
+        let mut invalid_values: Vec<OwnedElement> = vec![];
+        let mut valid_values: Vec<OwnedElement> = vec![];
+        let mut type_def: Option<TypeDefinitionImpl> = None;
+        let mut test_failed: bool = false;
+
+        for element in schema_content {
+            let annotations: Vec<&OwnedSymbolToken> = element.annotations().collect();
+            if annotations.contains(&&text_token("invalid_type")) {
+                // check for an invalid type validation
+                let type_def_result = schema_system.schema_type_from_element(&element, type_store);
+                if type_def_result.is_ok() {
+                    println!("ERROR: expected error for invalid type {:?}", element);
+                    test_failed = true;
+                    break;
+                }
+            } else if annotations.contains(&&text_token("invalid")) {
+                // get invalid values to validate
+                invalid_values = element
+                    .as_sequence()
+                    .unwrap()
+                    .iter()
+                    .map(|v| v.to_owned())
+                    .collect();
+            } else if annotations.contains(&&text_token("valid")) {
+                // get valid values to validate
+                valid_values = element
+                    .as_sequence()
+                    .unwrap()
+                    .iter()
+                    .map(|v| v.to_owned())
+                    .collect();
+            } else if annotations.contains(&&text_token("type")) {
+                // get type definition for type validation
+                type_def = Some(
+                    schema_system
+                        .schema_type_from_element(&element, type_store)
+                        .unwrap(),
+                );
+            } else {
+                continue;
+            }
+        }
+
+        match type_def {
+            None => {}
+            Some(schema_type) => {
+                for valid_value in valid_values {
+                    match schema_type.validate(&valid_value, type_store) {
+                        Ok(_) => {
+                            continue;
+                        }
+                        Err(error) => {
+                            println!("ERROR: {} for {:?}", error, valid_value);
+                            test_failed = true;
+                        }
+                    };
```

Comment:
I think this can be simplified.

```suggestion
                    if let Error(error) = schema_type.validate(&valid_value, type_store) {
                        println!("ERROR: {} for {:?}", error, valid_value);
                        test_failed = true;
                    };
```